### DISCUSSION TITLE
InvertOrNot POC

### DISCUSSION THREAD

**Discussion by ds2-jetten** [darkreader/darkreader#12503] [Ideas]
I was reading my TLDR; newsletter this morning and ran across this project that I thought might be of interest to this project. 
- [https://invertornot.com/](https://invertornot.com/)
- [https://github.com/mattismegevand/invertornot/](https://github.com/mattismegevand/invertornot/)

The API is free, but I have a feeling integrating the API directly into DR might not be the best idea for the developer who is making it available. However, perhaps you might be able to glean something from the source that could help improve image inversion.

**Comment by rich-ayr** [comment]
This works using a trained machine learning model. Since darkreader says on its README that even hosting the list through GitHub isn't possible due to high hosting cost. I don't know if they feel any better about downloading a 16 MB model for 100ms delayed response on whether to invert an image or not. Using the API directly is probably too costly or out of scope for this project (in case of putting your own API key in settings).

For an application like darkreader the development time would probably be spent better to train a custom model that also can detect/remove white backgrounds from non-transparent images or just in general is more geared toward the specific application.

_(NB: I'm not affiliated with darkreader or invertornot) and am merely giving my 2cts_
